File: AtCoder/ABC130/d.cpp
#include <iostream>

using namespace std;

int main()
{
    long N, K;
    cin >> N >> K;
    long a[N];
    for (int i = 0; i < N; i++) cin >> a[i];
    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < N; j++) dp[i][j] = 0;
    }
    dp[0][0] = a[0];
    for (int i = 1; i < N; i++) dp[0][i] = dp[0][i - 1] + a[i];
    for (int i = 1; i < N; i++)
    {
         for (int j = 1; j < N; j++) dp[i][j] = dp[i - 1][j] - a[i - 1];
    }
    long cnt = 0;
    for (int i = 0; i < N; i++)
    {
         for(int j = i; j < N; j++)
         {
             if (dp[i][j] >= K)
             {
                 cnt += N - (long)j;
                 break;
             }
         }
    }
    cout << cnt << "\n";
    return 0;
}

File: AtCoder/ABC130/d.cs
using System;
using System.Linq;
using System.Collections;
using System.Collections.Generic;

class Program
{
    public static void Main(string[] args)
    {
        string[] s = Console.ReadLine().Split(' ');
        long N = long.Parse(s[0]);
        long K = long.Parse(s[1]);
        long[] a = Array.ConvertAll(Console.ReadLine().Split(' '), long.Parse);
        long n = a[0];
        long cnt = 0;
        long res = 0;
        for (int i = 0; i < N; i++)
        {
            bool flag = false;
            if (i > 0) n -= a[i - 1];
            while (cnt < N)
            {
                if (cnt < i)
                {
                    cnt = i;
                    n = a[i];
                }
                if (flag) n += a[cnt];
                if (n >= K)
                {
                    res += N - cnt;
                    break;
                }
                cnt++;
                flag = true;
                if (cnt == N)
                {
                    cnt--;
                    break;
                }
            }
        }
        Console.WriteLine(res);
    }
}